clang/include/clang/Basic/DiagnosticCommonKinds.td

```diff
@@ -374,6 +374,8 @@ def err_ppc_impossible_musttail: Error<
   >;
 def err_aix_musttail_unsupported: Error<
   "'musttail' attribute is not supported on AIX">;
+def err_wasm_musttail_unsupported: Error<
+  "'musttail' attribute is not supported on this target without tail-call feature">;
```

Comment:
You shouldn't need to add a new diagnostic, we have automatic checking for whether the attribute is supported by the target (that's the code in Attr.td) which will diagnose the attribute as being unknown.